### PR TITLE
Refactor: cleanup codebase and improvements

### LINKED ISSUE #169: Clean up codebase
The number of algorithms are arranged/grouped into categories, first thing to do is to create submodules based on the categories. 

Work starts around next week. Going to assign myself.

### PR DESCRIPTION
**Pull request by uncomfyhalomacro**
Closes #169

### CODE REVIEW

File: .github/workflows/directory_workflow.yml [hunk 1]
```diff
@@ -1,46 +0,0 @@
-name: Directory/Filename Formatter workflow
```
Comment (Panquesito7) [inline .github/workflows/directory_workflow.yml:1]: Was this completely removed and not being used anymore?

Comment (uncomfyhalomacro) [reply .github/workflows/directory_workflow.yml:1]: This only adds complexity and causes annoyances during
a push and pull request. Removed. Doing this
manually is the best way. Last time i fetch and pulled to my local repository, it destroyed all my changes

Comment (uncomfyhalomacro) [reply .github/workflows/directory_workflow.yml:1]: I do think it's a useful tool but I think that should be done manually than adding it into an automation. We can direct contributors to the script.

File: .github/workflows/directory_workflow.yml [hunk 2]
```diff
@@ -1,46 +0,0 @@
...
-          done
-          git commit -am "Formatting filenames ${GITHUB_SHA::8}" || true
-      - name: Update DIRECTORY.md
-        run: |
-          wget https://raw.githubusercontent.com/TheAlgorithms/scripts/main/build_directory_md.py
-          python3 build_directory_md.py Julia ./src .jl TheAlgorithms.jl > DIRECTORY.md
-
-          git diff
-          git commit -m "Update DIRECTORY.md" DIRECTORY.md || true
-          git push --force origin HEAD:$GITHUB_REF || true
```
Comment (Panquesito7) [inline .github/workflows/directory_workflow.yml:46]: This seems to be causing the issue. If we push normally, there shouldn't be any issues.

Comment (uncomfyhalomacro) [reply .github/workflows/directory_workflow.yml:46]: Yes. That causes the issue but I don't think there is any point of removing those lines if it does not update the whole repository. That's why I suggest that we do it manually instead and point contributors to use the script from https://github.com/TheAlgorithms/scripts

**Review by Panquesito7 — APPROVED** [review]
Awesome. Thanks! 🚀

**Comment by Panquesito7** [comment]
Will be merging for now. If there's anything else to be done, feel free to make another PR. 🙂

File: .github/workflows/directory_workflow.yml (hunk 2 above)
Comment (Panquesito7) [reply .github/workflows/directory_workflow.yml:46]: I'll be creating a PR to have the filename formatter and directory workflow without causing any issues. You can see how it works in the C++ repository.

https://github.com/TheAlgorithms/C-Plus-Plus/blob/18f6cef2ad01c03cfffeefa0d4fa709e2145a79c/.github/workflows/awesome_workflow.yml#L25-L34

https://github.com/TheAlgorithms/C-Plus-Plus/blob/18f6cef2ad01c03cfffeefa0d4fa709e2145a79c/.github/workflows/awesome_workflow.yml#L89-L91

**Pull request merged**